src/routes/Admin/AdminView.js

```diff
@@ -13,31 +18,171 @@ class Admin extends Component {
     this.state = {
       products: [],
       isLading: false,
+      showModal: false,
+      showingLoadForms: false,
     };
-    this.clickItem = this.clickItem.bind(this);
+
+    this.navigateToItem = this.navigateToItem.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.handleCloseModalQuick = this.handleCloseModalQuick.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.createAddItemGlobal = this.createAddItemGlobal.bind(this);
+    this.handleCreate = this.handleCreate.bind(this);
   }
 
   async componentDidMount() {
     this.setState({ isLading: true });
     const productsJson = await fetch('/api/v1/products');
+    // console.log('need inf', productsJson)
     const products = await productsJson.json();
     this.setState({ products, isLading: false });
   }
 
-  clickItem = (evt, id) => {
+  handleOpenModal() {
+    this.setState({
+      showModal: true,
+      propsItem: undefined,
+      createNewItem: true,
+    });
+  }
+
+  handleCloseModal() {
+    this.setState({
+      showingLoadForms: true,
```

Comment:
why do you still showing that loading when the modal is gonna close?

Reply:
because I need to call 'Loading...' for 3 seconds, which will close itself

Reply:
didn't get it. why? you should show the loading for the time of updating, not just with some random timeout.